scratch-js/Sprite.mjs

```diff
@@ -542,6 +595,42 @@ export class Sprite extends SpriteBase {
     }
   }
 
+  nearestEdge() {
+      const bounds = this._project.renderer.getBoundingBox(this);
+      const {width: stageWidth, height: stageHeight} = this.stage;
+      const distLeft = Math.max(0, (stageWidth / 2) + bounds.left);
+      const distTop = Math.max(0, (stageHeight / 2) - bounds.top);
+      const distRight = Math.max(0, (stageWidth / 2) - bounds.right);
+      const distBottom = Math.max(0, (stageHeight / 2) + bounds.bottom);
+      // Find the nearest edge.
+      let nearestEdge = '';
+      let minDist = Infinity;
+      if (distLeft < minDist) {
+        minDist = distLeft;
+        nearestEdge = Sprite.Edge.LEFT;
+      }
+      if (distTop < minDist) {
+        minDist = distTop;
+        nearestEdge = Sprite.Edge.TOP;
+      }
+      if (distRight < minDist) {
+        minDist = distRight;
+        nearestEdge = Sprite.Edge.RIGHT;
+      }
+      if (distBottom < minDist) {
+        minDist = distBottom;
+        nearestEdge = Sprite.Edge.BOTTOM;
+      }
+      if (minDist > 0) {
+        nearestEdge = null;
+      }
+      return nearestEdge;
+  }
+
+  touchingEdge() {
+    return !!this.nearestEdge();
+  }
```

Comment:
Let's move this to inside the `touching` method that already exists. It can be a new case (`case "edge"`) within the switch statement.